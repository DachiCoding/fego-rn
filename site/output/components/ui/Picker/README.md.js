webpackJsonp([19],{486:function(t,e){t.exports={content:["article",["h1","\u9009\u62e9\u5668 Picker"],["h3","\u7ec4\u4ef6\u63cf\u8ff0"],["ul",["li",["p","\u6a21\u6001\u91cc\u7684\u9009\u62e9\u5668"]]],["h3","\u793a\u4f8b\u4ee3\u7801"],["pre",{lang:"html",highlighted:"import Picker from 'somewhere'\n&lt;Picker\n\tdata={pickerData}\n\tvisible={this.state.pickerVisible}\n\tonClose={()=>{this.setState({\n\t\tpickerVisible: false\n\t})}}\n\tmaskClosable={false}\n\t{...otherProps}\n/>"},["code","import Picker from 'somewhere'\n<Picker\n\tdata={pickerData}\n\tvisible={this.state.pickerVisible}\n\tonClose={()=>{this.setState({\n\t\tpickerVisible: false\n\t})}}\n\tmaskClosable={false}\n\t{...otherProps}\n/>"]],["h3","\u8bf4\u660e"],["blockquote",["p","Picker \u7684\u5185\u90e8\u5b9e\u73b0\u8fd8\u662f\u5f15\u7528\u4e86 PickerView, \u53ea\u662f\u5c06 PickerView \u653e\u5728\u4e86\u6a21\u6001\u91cc\uff0c\u56e0\u6b64\u5176\u90e8\u5206 props (eg: cascade, cols, initialValue, onChange) \u4ee5\u53ca\u6570\u636e\u6e90 data \u7684\u542b\u4e49\u90fd\u4e0e PickerView \u4e00\u81f4"]],["h2","API"],["h3","props\u5217\u8868"],["table",["thead",["tr",["th","\u5c5e\u6027"],["th","\u8bf4\u660e"],["th","\u7c7b\u578b"],["th","\u9ed8\u8ba4\u503c"]]],["tbody",["tr",["td","data"],["td","\u9009\u62e9\u5668\u7684\u6570\u636e\u6e90"],["td","array"],["td",["span"," "]]],["tr",["td","cascade"],["td","\u9009\u62e9\u5668\u662f\u5426\u7ea7\u8054"],["td","bool"],["td","false"]],["tr",["td","cols"],["td","\u5217\u6570,\u7ea7\u8054\u5f0f\u65f6\u6709\u7684\u7ea7\u6570"],["td","number"],["td","-"]],["tr",["td","initialValue"],["td","picker \u521d\u59cb\u503c\uff0c\u683c\u5f0f\u4e3a",["span","v1, v2, v3"],"\uff0c\u5bf9\u5e94\u6570\u636e\u6e90\u7684\u76f8\u5e94\u7ea7\u5c42value"],["td","array"],["td",["span"," "]]],["tr",["td","onChange"],["td","\u6bcf\u5217\u6570\u636e\u9009\u62e9\u53d8\u5316\u540e\u7684\u56de\u8c03\u51fd\u6570"],["td","Function(selectedValue, selectedIndex, selectedLabel){}"],["td","-"]],["tr",["td","visible"],["td","\u662f\u5426\u53ef\u89c1"],["td","bool"],["td","false"]],["tr",["td","title"],["td","\u6a21\u6001\u5185\u6807\u9898"],["td","string"],["td","\u8bf7\u9009\u62e9"]],["tr",["td","okText"],["td","\u786e\u5b9a\u7684\u6587\u6848"],["td","string"],["td","\u786e\u5b9a"]],["tr",["td","cancelText"],["td","\u53d6\u6d88\u7684\u6587\u6848"],["td","string"],["td","\u53d6\u6d88"]],["tr",["td","maskClosable"],["td","\u70b9\u51fb\u8499\u5c42\u662f\u5426\u5141\u8bb8\u5173\u95ed"],["td","string"],["td","true"]],["tr",["td","onClose"],["td","\u5173\u95ed\u5f39\u6846\u7684\u56de\u8c03\u51fd\u6570"],["td","Function(){}"],["td","-"]],["tr",["td","onConfirm"],["td","\u786e\u5b9a\u6309\u94ae\u7684\u56de\u8c03\u51fd\u6570"],["td","Function(selectedValue, selectedIndex, selectedLabel){}"],["td","-"]]]],["h3","\u6837\u5f0f\u5bf9\u8c61styles"],["table",["thead",["tr",["th","\u5c5e\u6027"],["th","\u8bf4\u660e"],["th","\u9002\u7528\u7c7b\u578b"]]],["tbody"]],["h3","props data \u8bf4\u660e"],["h4","data \u7684\u6570\u636e\u683c\u5f0f\u4e3a ",["code","Array<{value, label}>"]," \u6216 ",["code","Array<Array<{value, label}>>"]],["ul",["li",["p","PivkerView \u53ea\u542b\u4e00\u5217\u6570\u636e"]]],["pre",{lang:"html",highlighted:"[{\n\tlabel: '2011\u5e74',\n\tvalue: '2011'\n}, {\n\tlabel: '2012\u5e74',\n\tvalue: '2012'\n}]"},["code","[{\n\tlabel: '2011\u5e74',\n\tvalue: '2011'\n}, {\n\tlabel: '2012\u5e74',\n\tvalue: '2012'\n}]"]],["ul",["li",["p","PivkerView \u542b\u4e24\u5217\u6570\u636e"]]],["pre",{lang:"html",highlighted:"[\n\t[{\n\t\tlabel: '2011\u5e74',\n\t\tvalue: '2011'\n\t}, {\n\t\tlabel: '2012\u5e74',\n\t\tvalue: '2012'\n\t}],\n\n\t[{\n\t\tlabel: '1\u6708',\n\t\tvalue: '1'\n\t}, {\n\t\tlabel: '2\u6708',\n\t\tvalue: '2'\n\t}]\n]"},["code","[\n\t[{\n\t\tlabel: '2011\u5e74',\n\t\tvalue: '2011'\n\t}, {\n\t\tlabel: '2012\u5e74',\n\t\tvalue: '2012'\n\t}],\n\n\t[{\n\t\tlabel: '1\u6708',\n\t\tvalue: '1'\n\t}, {\n\t\tlabel: '2\u6708',\n\t\tvalue: '2'\n\t}]\n]"]],["h4","\u6bcf\u9879\u6570\u636e\u9664\u4e86 label \u548c value \u4e24\u4e2a\u5c5e\u6027\u5916\uff0c\u8fd8\u6709\u7b2c\u4e09\u4e2a\u5c5e\u6027\uff1achildren\uff0c\u8868\u793a\u7ea7\u8054\u65f6\u8be5\u9879\u503c\u4e0b\u4e00\u7ea7\u7684\u6570\u636e"],["pre",{lang:"html",highlighted:"[{\n\tlabel: '\u5ddd\u83dc',\n\tvalue: 'chuancai',\n\tchildren: [{\n\t\tlabel: '\u5ddd\u83dc-\u706b\u9505',\n\t\tvalue: 'huoguo',\n\t\tchildren: [{\n\t\t\tlabel: '\u5ddd\u83dc-\u706b\u9505-\u6bdb\u809a',\n\t\t\tvalue: 'maodu'\n\t\t}, {\n\t\t\tlabel: '\u5ddd\u83dc-\u706b\u9505-\u9e2d\u80a0',\n\t\t\tvalue: 'jiachang'\n\t\t}]\n\t}, {\n\t\tlabel: '\u5ddd\u83dc-\u5bb6\u5e38',\n\t\tvalue: 'jiachang',\n\t\tchildren: [{\n\t\t\tlabel: '\u5ddd\u83dc-\u5bb6\u5e38-\u5bab\u4fdd\u9e21\u4e01',\n\t\t\tvalue: 'jiding'\n\t\t}, {\n\t\t\tlabel: '\u5ddd\u83dc-\u5bb6\u5e38-\u6c34\u716e\u9c7c',\n\t\t\tvalue: 'yu'\n\t\t}]\n\t}]\n}, {\n\tlabel: '\u53f0\u6e7e',\n\tvalue: 'taiwan',\n\tchildren: [{\n\t\tlabel: '\u53f0\u6e7e-\u5c0f\u5403',\n\t\tvalue: 'xiaochi',\n\t\tchildren: [{\n\t\t\tlabel: '\u53f0\u6e7e-\u5c0f\u5403-\u5364\u8089\u996d',\n\t\t\tvalue: 'lurou'\n\t\t}, {\n\t\t\tlabel: '\u53f0\u6e7e-\u5c0f\u5403-\u80a5\u8089\u996d',\n\t\t\tvalue: 'feirou'\n\t\t}]\n\t}, {\n\t\tlabel: '\u53f0\u6e7e-\u70e7\u70e4',\n\t\tvalue: 'shao',\n\t\tchildren: []\n\t}]\n}]"},["code","[{\n\tlabel: '\u5ddd\u83dc',\n\tvalue: 'chuancai',\n\tchildren: [{\n\t\tlabel: '\u5ddd\u83dc-\u706b\u9505',\n\t\tvalue: 'huoguo',\n\t\tchildren: [{\n\t\t\tlabel: '\u5ddd\u83dc-\u706b\u9505-\u6bdb\u809a',\n\t\t\tvalue: 'maodu'\n\t\t}, {\n\t\t\tlabel: '\u5ddd\u83dc-\u706b\u9505-\u9e2d\u80a0',\n\t\t\tvalue: 'jiachang'\n\t\t}]\n\t}, {\n\t\tlabel: '\u5ddd\u83dc-\u5bb6\u5e38',\n\t\tvalue: 'jiachang',\n\t\tchildren: [{\n\t\t\tlabel: '\u5ddd\u83dc-\u5bb6\u5e38-\u5bab\u4fdd\u9e21\u4e01',\n\t\t\tvalue: 'jiding'\n\t\t}, {\n\t\t\tlabel: '\u5ddd\u83dc-\u5bb6\u5e38-\u6c34\u716e\u9c7c',\n\t\t\tvalue: 'yu'\n\t\t}]\n\t}]\n}, {\n\tlabel: '\u53f0\u6e7e',\n\tvalue: 'taiwan',\n\tchildren: [{\n\t\tlabel: '\u53f0\u6e7e-\u5c0f\u5403',\n\t\tvalue: 'xiaochi',\n\t\tchildren: [{\n\t\t\tlabel: '\u53f0\u6e7e-\u5c0f\u5403-\u5364\u8089\u996d',\n\t\t\tvalue: 'lurou'\n\t\t}, {\n\t\t\tlabel: '\u53f0\u6e7e-\u5c0f\u5403-\u80a5\u8089\u996d',\n\t\t\tvalue: 'feirou'\n\t\t}]\n\t}, {\n\t\tlabel: '\u53f0\u6e7e-\u70e7\u70e4',\n\t\tvalue: 'shao',\n\t\tchildren: []\n\t}]\n}]"]]],meta:{title:"Picker",subTitle:"\u9009\u62e9\u5668",filename:"components/ui/Picker/README.md"},toc:["ul",["li",["a",{className:"bisheng-toc-h1",href:"#\u9009\u62e9\u5668-Picker",title:"\u9009\u62e9\u5668 Picker"},"\u9009\u62e9\u5668 Picker"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u7ec4\u4ef6\u63cf\u8ff0",title:"\u7ec4\u4ef6\u63cf\u8ff0"},"\u7ec4\u4ef6\u63cf\u8ff0"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u793a\u4f8b\u4ee3\u7801",title:"\u793a\u4f8b\u4ee3\u7801"},"\u793a\u4f8b\u4ee3\u7801"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u8bf4\u660e",title:"\u8bf4\u660e"},"\u8bf4\u660e"]],["li",["a",{className:"bisheng-toc-h2",href:"#API",title:"API"},"API"]],["li",["a",{className:"bisheng-toc-h3",href:"#props\u5217\u8868",title:"props\u5217\u8868"},"props\u5217\u8868"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u6837\u5f0f\u5bf9\u8c61styles",title:"\u6837\u5f0f\u5bf9\u8c61styles"},"\u6837\u5f0f\u5bf9\u8c61styles"]],["li",["a",{className:"bisheng-toc-h3",href:"#props-data-\u8bf4\u660e",title:"props data \u8bf4\u660e"},"props data \u8bf4\u660e"]],["li",["a",{className:"bisheng-toc-h4",href:"#data-\u7684\u6570\u636e\u683c\u5f0f\u4e3a-Array<{value,-label}>-\u6216-Array<Array<{value,-label}>>",title:"data \u7684\u6570\u636e\u683c\u5f0f\u4e3a Array<{value, label}> \u6216 Array<Array<{value, label}>>"},"data \u7684\u6570\u636e\u683c\u5f0f\u4e3a Array<{value, label}> \u6216 Array<Array<{value, label}>>"]],["li",["a",{className:"bisheng-toc-h4",href:"#\u6bcf\u9879\u6570\u636e\u9664\u4e86-label-\u548c-value-\u4e24\u4e2a\u5c5e\u6027\u5916\uff0c\u8fd8\u6709\u7b2c\u4e09\u4e2a\u5c5e\u6027\uff1achildren\uff0c\u8868\u793a\u7ea7\u8054\u65f6\u8be5\u9879\u503c\u4e0b\u4e00\u7ea7\u7684\u6570\u636e",title:"\u6bcf\u9879\u6570\u636e\u9664\u4e86 label \u548c value \u4e24\u4e2a\u5c5e\u6027\u5916\uff0c\u8fd8\u6709\u7b2c\u4e09\u4e2a\u5c5e\u6027\uff1achildren\uff0c\u8868\u793a\u7ea7\u8054\u65f6\u8be5\u9879\u503c\u4e0b\u4e00\u7ea7\u7684\u6570\u636e"},"\u6bcf\u9879\u6570\u636e\u9664\u4e86 label \u548c value \u4e24\u4e2a\u5c5e\u6027\u5916\uff0c\u8fd8\u6709\u7b2c\u4e09\u4e2a\u5c5e\u6027\uff1achildren\uff0c\u8868\u793a\u7ea7\u8054\u65f6\u8be5\u9879\u503c\u4e0b\u4e00\u7ea7\u7684\u6570\u636e"]]]}}});