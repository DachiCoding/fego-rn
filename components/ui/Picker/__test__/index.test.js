import React from 'react'
import Renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Picker from '../index'

describe('Picker Tests', () => {
	let data, handler;
	beforeEach(() => {
		data = [{
			label: 'item1',
			value: '1'
		}, {
			label: 'item2',
			value: '2'
		}, {
			label: 'item3',
			value: '3'
		}, {
			label: 'item4',
			value: '4'
		}, {
			label: 'item5',
			value: '5'
		}]
	})
	
	it('renders correctly', () => {
		const tree = Renderer.create(<Picker
			data={data}
			visible={true}
		/>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('default props tests', () => {
		const wrapper = shallow(<Picker
			data={data}
			visible={true}
		/>);
		
		expect(wrapper.instance().props.title).toBe('请选择');
		expect(wrapper.instance().props.okText).toBe('确定');
		expect(wrapper.instance().props.cancelText).toBe('取消');
	});

	it('close picker event', () => {
		let handler = jest.fn();
		const wrapper = shallow(<Picker
			data={data}
			visible={true}
			onClose={handler}
		/>);
		// 找到取消按钮
		wrapper.childAt(0).childAt(0).childAt(0).simulate('press');
		expect(handler).toHaveBeenCalled();
	});
})


