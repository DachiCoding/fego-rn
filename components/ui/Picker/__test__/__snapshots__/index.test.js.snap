// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker Tests renders correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "flex": 1,
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "black",
              "opacity": 0.5,
            },
          ]
        }
      />
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "white",
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            false,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "#ccc",
              "borderBottomWidth": 1,
              "flex": 0,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-start",
                "justifyContent": "center",
                "opacity": 1,
                "paddingHorizontal": 20,
                "paddingVertical": 10,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#108ee9",
                    "fontSize": 16,
                  },
                  Object {},
                ]
              }
            >
              取消
            </Text>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            请选择
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "justifyContent": "center",
                "opacity": 1,
                "paddingHorizontal": 20,
                "paddingVertical": 10,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#108ee9",
                    "fontSize": 16,
                  },
                  Object {},
                ]
              }
            >
              确定
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "flex-start",
              }
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "item1",
                    "textColor": undefined,
                    "value": "1",
                  },
                  Object {
                    "label": "item2",
                    "textColor": undefined,
                    "value": "2",
                  },
                  Object {
                    "label": "item3",
                    "textColor": undefined,
                    "value": "3",
                  },
                  Object {
                    "label": "item4",
                    "textColor": undefined,
                    "value": "4",
                  },
                  Object {
                    "label": "item5",
                    "textColor": undefined,
                    "value": "5",
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
